lib/pagseguro/transaction_request/response.rb

```diff
@@ -1,30 +1,26 @@
 module PagSeguro
   class TransactionRequest
     class Response
-      extend Forwardable
-
-      def_delegators :response, :success?
-      attr_reader :response
-
       def initialize(response)
         @response = response
       end
 
-      def errors
-        @errors ||= Errors.new(response)
+      def serialize
+        if success?
+          xml = Nokogiri::XML(response.body).css("transaction").first
+          ResponseSerializer.new(xml).serialize
+        else
+          { errors: Errors.new(response) }
+        end
       end
 
-      def code
-        @code ||= response.data.css("transaction > code").text if success?
+      def success?
+        (response.success? && response.xml?) ? true : false
```

Comment:
@adbatista I'm not sure if `!!response.xml?` is more readable. seems like a hack.